### PR TITLE
add new cut float feature transform

### PR DESCRIPTION
**Pull request by jq**
@deerzq @hectorgon

### CODE REVIEW

File: core/src/main/java/com/airbnb/aerosolve/core/transforms/CutFloatFeatureTransform.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package com.airbnb.aerosolve.core.transforms;
+
+import com.airbnb.aerosolve.core.FeatureVector;
+import com.airbnb.aerosolve.core.util.Util;
+import com.typesafe.config.Config;
+
+import java.util.List;
+import java.util.Map;
+
+public class CutFloatFeatureTransform extends Transform {
```
Comment (hectorgon) [inline core/src/main/java/com/airbnb/aerosolve/core/transforms/CutFloatFeatureTransform.java]: Add a comment as to what this does

**Comment by hectorgon** [comment]
:+1:

File: core/src/main/java/com/airbnb/aerosolve/core/transforms/CutFloatFeatureTransform.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    private double lowerBound;
+    private double upperBound;
+    private String outputName; // output family name, if not specified, output to fieldName1
+
+    @Override
+    public void configure(Config config, String key) {
+        fieldName1 = config.getString(key + ".field1");
+        keys = config.getStringList(key + ".keys");
+        lowerBound = config.getDouble(key + ".lower_bound");
+        upperBound = config.getDouble(key + ".upper_bound");
```
Comment (deerzq) [inline core/src/main/java/com/airbnb/aerosolve/core/transforms/CutFloatFeatureTransform.java]: what if people only want one-sided cut?

File: core/src/main/java/com/airbnb/aerosolve/core/transforms/CutFloatFeatureTransform.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.Map;
+
+/*
+  remove features larger than upperBound or smaller than lowerBound
+ */
+public class CutFloatFeatureTransform extends Transform {
+    private String fieldName1;
+    private List<String> keys;
+    private double lowerBound;
+    private Double upperBound;
```
Comment (deerzq) [inline core/src/main/java/com/airbnb/aerosolve/core/transforms/CutFloatFeatureTransform.java]: `double`?

Comment (jq) [inline core/src/main/java/com/airbnb/aerosolve/core/transforms/CutFloatFeatureTransform.java]: nice catch! updated!

File: core/src/main/java/com/airbnb/aerosolve/core/transforms/CutFloatFeatureTransform.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        Map<String, Double> feature1 = floatFeatures.get(fieldName1);
+        if (feature1 == null) {
+            return;
+        }
+        Map<String, Double> feature2 = Util.getOrCreateFloatFeature(outputName, floatFeatures);
+        for (String key : keys) {
+            Double v = feature1.get(key);
+            if (v != null) {
+                if (v > upperBound || v < lowerBound) {
+                    feature2.remove(key);
```
Comment (deerzq) [inline core/src/main/java/com/airbnb/aerosolve/core/transforms/CutFloatFeatureTransform.java]: if feature2 is a newly created map, there may not be `key` in it, then you cannot remove it.

Comment (deerzq) [inline core/src/main/java/com/airbnb/aerosolve/core/transforms/CutFloatFeatureTransform.java]: I guess it does not throw exception in this case, but perhaps add a comment or do a check before removing ...

**Comment by deerzq** [comment]
Some nits, otherwise LGTM.

File: core/src/main/java/com/airbnb/aerosolve/core/transforms/CutFloatFeatureTransform.java (hunk 4 above)
Comment (jq) [inline core/src/main/java/com/airbnb/aerosolve/core/transforms/CutFloatFeatureTransform.java]: Thanks for review!  updated with check first

**Comment by hectorgon** [comment]
Thanks for the great code review @deerzq I am glad you caught those bugs

**Pull request merged**